### PR TITLE
[corechecks/snmp] Add additional metadata fields

### PR DESCRIPTION
**Pull request by AlexandreYang**
### What does this PR do?

[corechecks/snmp] Add additional metadata

### Motivation

Adding new metadata fields that can be defined in profiles.

Example of usage in profile: https://github.com/DataDog/integrations-core/pull/10667

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/collector/corechecks/snmp/test/metadata_conf.d/snmp.d/profiles/_base.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Base profile that should only contain any items we want to provide for all profiles.
+
+metric_tags:
+  - OID: 1.3.6.1.2.1.1.5.0
+    symbol: sysName
+    tag: snmp_host
+
+metadata:
```
Comment (gmarabout) [inline pkg/collector/corechecks/snmp/test/metadata_conf.d/snmp.d/profiles/_base.yaml:8]: 👍🏻

**Review by gmarabout — APPROVED** [review]
Very nice!

File: pkg/collector/corechecks/snmp/checkconfig/config_validate_enrich.go [hunk 2]
```diff
@@ -10,8 +10,15 @@ var validMetadataResources = map[string]map[string]bool{
 		"name":          true,
 		"description":   true,
 		"sys_object_id": true,
+		"location":      true,
```
Comment (pducolin) [inline pkg/collector/corechecks/snmp/checkconfig/config_validate_enrich.go:13]: Why `ip_address` and `network_address` are missing here?

**Review by pducolin — APPROVED** [review]

File: pkg/collector/corechecks/snmp/checkconfig/config_validate_enrich.go (hunk 2 above)
Comment (AlexandreYang) [reply pkg/collector/corechecks/snmp/checkconfig/config_validate_enrich.go:13]: The `ip_address` and `subnet` (do you mean `device.subnet` field?) are not based on profiles definition, so no need to validate them here and they can't be overwritten via profile.

- ip_address is based on instance ip_address
- subnet is from autodiscovery

**Comment by AlexandreYang** [comment]
rebased onto
```
git rebase --onto main alex/snmp_profile_metadata
```

**Pull request merged**